basculehttp/middleware_examples_test.go

```diff
@@ -55,3 +56,111 @@ func ExampleMiddleware_basicauth() {
 	// principal: joe
 	// with basic auth response code: 200
 }
+
+// ExampleMiddleware_authentication shows how to authenticate a token.
+func ExampleMiddleware_authentication() {
+	tp, _ := NewAuthorizationParser(
+		WithBasic(),
+	)
+
+	m, _ := NewMiddleware(
+		UseAuthenticator(
+			NewAuthenticator(
+				bascule.WithTokenParsers(tp),
+				bascule.WithValidators(
+					AsValidator(
+						// the signature of validator closures is very flexible
+						// see bascule.AsValidator
+						func(token bascule.Token) error {
+							if basic, ok := token.(BasicToken); ok && basic.Password() == "correct_password" {
+								return nil
+							}
+
+							return bascule.ErrBadCredentials
```

Comment:
I had to use bascule.ErrInvalidCredentials to compile

Reply:
from these two comments, it sounds like you haven't checked out the branch properly.  these are changes made in this PR.  the unit tests pass, so we know the branch compiles on its own, with no changes.